backend/app/routers/experiment_templates.py

```diff
@@ -161,3 +275,24 @@ def find_specific_experiment_templates(
         return ExperimentTemplate.find(multi_query, **page_kwargs)
 
     return ExperimentTemplate.find_all(**page_kwargs)
+
+
+async def get_experiment_template_if_accessible_or_raise(
+    template_id: PydanticObjectId, user: dict | None, write_access: bool = False
+) -> ExperimentTemplate:
+    template = await ExperimentTemplate.get(template_id)
+    if template is None:
```

Comment:
1. All the condition cases end with either `return` or `raise`, but they are written in a way that looks like they are independent. I recommend rewriting them in a `if-elif-else` form to make it clear and avoid bugs later on.
2. I think it would be better if we just either return the template or state that user does not have access. I think we don't need to share the extra information that the experiment exists.

Reply:
Regarding the point 2: This `get_experiment_template_if_accessible_or_raise` function is called in majority of endpoints and if we don't check for whether the template actually exists or now, we would need to check this elsewhere. I suppose, to keep the logic separate, we could add another function that checks whether the template actually exists or not and then we can use it in combination with this function in question, but I feel like it makes it more messy, I suppose.

Reply:
Oh, sorry for the confusion. I meant keeping the logic just unifying the error message. Possibly something like this:
```python
    template = await ExperimentTemplate.get(template_id)

    access_denied_error = HTTPException(
        status_code=status.HTTP_401_UNAUTHORIZED,
        detail="You cannot access this experiment template.",
    )

    if template is None:
        raise access_denied_error
    else:
        if write_access is False and template.is_public:
            # Public templates are readable to everyone
            return template
        elif user is not None and template.created_by == user["email"]:
            # TODO: Add experiment access management
            return template
        else:
            raise access_denied_error
```